### ISSUE TITLE
Return only the first validation error

### ISSUE THREAD

**Issue by JaredCorduan** [IntersectMBO/cardano-ledger#1431]
The shelley ledger is currently returning all the validation errors when a block/transaction is invalid. This can be confusing, though, in cases where one error is really the cause of the others. For example, if you supply a  bad transaction input, the `ValueNotConservedUTxO` error is almost certainly also going to trigger.

We can start to make this clearer by reporting only the first error.

**Comment by tigrpoolcom** [comment]
Hello together,
i've run into this error today and tbh it was a real pain in the ass. Please look into it.
I am currently testing to register and deregister a pool on the testnet with the V1.25.1 version.

Steps to reproduce:
1. Create a pool-reg cert (and dependencies)
2. Register the pool-reg cert
3. Generate a new pool-reg cert with changed fees (i changed from 1 to 0.15)
4. Then try to register the new pool-reg on the blockchain 

-> You will now encounter an error saying that you do not have enough funds, even though it's showing you have enough funds. This is really annoying, please fix this and instead return an error message like "Pool is already registered".

**Here my log:**

Executed Command: /bin/bash -c "cardano-cli transaction build-raw --tx-in 37e2e7cb47e685aa4efc170f298ab7d80c9ba354b4e29b8bf2e2219864d54c83#0 --tx-out $(cat payment.addr)+98065927 --invalid-hereafter 19310234 --fee 195377 --certificate-file pool-registration.cert --certificate-file delegation.cert --out-file tx.raw;"
Executed Command: /bin/bash -c "cardano-cli transaction sign --tx-body-file tx.raw --signing-key-file payment.skey --signing-key-file stake.skey --signing-key-file cold.skey --out-file tx.signed;"
Executed Command: /bin/bash -c "cardano-cli transaction submit --tx-file tx.signed --$CDN_NETCMD ;"
Error Result: Shelley command failed: transaction submit Error: Error while submitting tx: ApplyTxError [LedgerFailure (UtxowFailure (UtxoFailure (ValueNotConservedUTxO (Value 598261304 (fromList [])) (Value 98261304 (fromList [])))))]

**Comment by tigrpoolcom** [comment]
Update: The error message only pops up if i send a pool-reg.cert with the deducted poolDeposit fee of 500ADA. I then receive this error message. When i sent the changes certificate with a normal TX fee it is working.

**Comment by JaredCorduan** [comment]
hello @tigrpoolcom,

In this case, the error message is actually being informative. The error was not saying that you do not have enough funds, but rather that the transaction does not preserve the total lovelace in the system. More specifically, it is saying that `598261304` lovelace are being consumed, while only `98261304` lovelace are being produced.

The error is not that the pool is already registered, but that the transaction is overpaying by including enough funds to cover a the stake pool deposit which has already been paid (`598261304 - 98261304 = 500000000 = stake pool deposit`).

Happy staking!

**Comment by JaredCorduan** [comment]
this is related to https://github.com/input-output-hk/cardano-ledger/issues/1304

**Comment by lehins** [comment]
I do not think it is a good idea in general to short-circuit on the very first predicate failure, because there could be many legitimate and unrelated reason why a transaction is invalid.
For example invalid spending witness and attempt to unregister a stake credential that does not exist. We don't want the user to resubmit a transaction for every problem that they fix during transaction building. Moreover, there is no guarantee that the first predicate failure that appears in the list is indeed the actual culprit of the transaction validation failure. 

Short circuiting is useful in some cases as an optimization, in particular during transaction revalidation in the mempool (see #4849), which we are planning to implement. But from the perspective of the user submitting a transaction it is better to provide as much information as possible, instead of trying to decide which is the most useful predicate failure.

I think this problem should be solved with some really good documentation of all of the predicate failures that ledger can produce. There is even already a ticket for it: #2754

I am closing this ticket as won't fix. For anyone still feeling like this would be a useful feature, please, feel free to reopen this ticket.
